dag.go

```diff
@@ -102,6 +109,152 @@ func (m *Manifest) RootCID() cid.Cid {
 // 	return nodes
 // }
 
+// SubDAG returns a Manifest of a DAG at a specific hash in an already existing Manifest
+func SubDAG(m *Manifest, id string) (*Manifest, error) {
+	info := &Info{
+		Manifest: m,
+	}
+	root, err := m.IDIndex(id)
+	if err != nil {
+		return nil, err
+	}
+	converter, err := newSubDAGConverter(info, root)
+	if err != nil {
+		return nil, err
+	}
+	converter.Convert()
+	return converter.Manifest, nil
+}
+
+func newSubDAGConverter(prevInfo *Info, root int) (*subDAGConverter, error) {
+	if prevInfo.Manifest == nil {
+		return nil, fmt.Errorf("no manifest provided")
+	}
+	if root < 0 || root > len(prevInfo.Manifest.Nodes)-1 {
+		return nil, ErrIndexOutOfRange
+	}
+	m := prevInfo.Manifest
+	PrevLabels := map[int]string{}
+	for path, index := range prevInfo.Labels {
+		PrevLabels[index] = path
+	}
+
+	lastLink := m.Links[len(m.Links)-1]
+	return &subDAGConverter{
+		Conversion:         map[int]int{},
+		PrevManifest:       m,
+		Manifest:           &Manifest{Nodes: []string{}, Links: [][2]int{}},
```

Comment:
No need to allocate arrays here:
```suggestion
		Manifest:           &Manifest{},
```